src/flisp/read.c

```diff
@@ -44,11 +46,15 @@ int isnumtok_base(fl_context_t *fl_ctx, char *tok, value_t *pval, int base)
             if (pval) *pval = mk_double(fl_ctx, d);
             return 1;
         }
+
+        // NOTE(#49689): DO NOT convert double to float directly,
+        //  indirect conversion (string->double->float) will lose precision.
+        f = jl_strtof_c(tok, &end);
```

Comment:
This call (plus the call to `jl_strtod_c` above) can be sunk inside the conditional.

Reply:
I moved `jl_strtof_c` into the branch of `if (pval)`.
But left `jl_strtod_c` untouched.

https://github.com/JuliaLang/julia/blob/e204e200df8f57e516920c056b6ec2eb37d34079/src/flisp/read.c#L42-L46
Calling `jl_strtod_c` updates `char *end` before we can do dereference `*end`.
So I don't think we can move `jl_strtod_c`.